test/sim/moves/swallow.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+const assert = require("./../../assert");
+const common = require("./../../common");
+
+let battle;
+
+describe("Swallow", function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it("should heal 1/4 health (rounded down) after 1 stockpile", function () {
+		battle = common.createBattle([
+			[
+				{
+					species: "Seviper",
+					ability: "shedskin",
+					moves: ["stockpile", "swallow"],
+				},
+			],
+			[{species: "Zangoose", ability: "immunity", moves: ["sleeptalk", "falseswipe"]}],
+		]);
+
+		battle.makeChoices("move stockpile", "move falseswipe");
```

Comment:
You can use something like Seismic Toss instead so HP is fixed throughout the whole process.